lib/EventDispatcher/DebugEventDispatcher.php

```diff
@@ -23,6 +23,11 @@
 class DebugEventDispatcher extends ContainerAwareEventDispatcher
 {
     /**
+     * @var int
+     */
+    private $depth = 0;
```

Comment:
What if the `doDispatch` level exits early, e.g. because of an exception? I don't see where this gets reset afterwards.